### ISSUE TITLE
test_busycursor.py segfaults on Linux PySide6 6.3, 6.4

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#2520]
### Short description

test_busycursor.py segfaults under the following conditions
* on Linux PySide6 6.3, 6.4 *and*
* *when executed alone*

### Code to reproduce

```python
pytest tests/widgets/test_busycursor.py
```

### Real behavior

```
(py310_pyside63) :~/repos/pyqtgraph$ pytest tests/widgets/test_busycursor.py
============================================== test session starts ===============================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/user/repos/pyqtgraph, configfile: pytest.ini
collected 1 item

tests/widgets/test_busycursor.py .                                                                         [100%]

=============================================== 1 passed in 0.07s ================================================
Segmentation fault
```

### Tested environment(s)

 * PyQtGraph version: master 
 * Qt Python binding: PySide6 6.4.0.1, 6.3.2
 * Python version: 3.10.6
 * NumPy version: 1.23.4 
 * Operating system: Ubuntu 22.04
 * Installation method: pip 

### Additional context
When tested on the CI running just this test, it does not happen on macOS and Windows, nor on Linux with PySide 5.15, 6.2.4

While debugging the QMenu leak issue in #2497, various segfaults at end of *successful* pytest run could be triggered on Linux PySide6 >= 6.3. This may be related.

**Comment by pijyoi** [comment]
A MWE would be the following
```python
from PySide6 import QtGui
app = QtGui.QGuiApplication([])
t = app.thread()
```
i.e. the program will segfault on program end if you take the gui thread object.

**Comment by pijyoi** [comment]
Looks like a bug report got filed https://bugreports.qt.io/browse/PYSIDE-2254 and a fix is pending https://codereview.qt-project.org/c/pyside/pyside-setup/+/465817

**Comment by pijyoi** [comment]
The fix has been merged to 6.4 https://code.qt.io/cgit/pyside/pyside-setup.git/commit/?h=6.4&id=4ee2bfc29b9367f14a60a6b81a83bd92d7c24f25
So the skipped tests can probably be reinstated when 6.4.3 gets released.
